src/engine/strat_engine/backstore/crypt.rs

```diff
@@ -583,6 +544,101 @@ impl CryptHandle {
     }
 }
 
+fn setup_crypt_handle(
+    physical_path: &Path,
+    unlock_method: Option<UnlockMethod>,
+) -> Result<Option<CryptHandle>> {
+    let device_result = device_from_physical_path(physical_path);
+    let mut device = match device_result {
+        Ok(None) => return Ok(None),
+        Ok(Some(mut dev)) => {
+            if !is_encrypted_stratis_device(&mut dev) {
+                return Ok(None);
+            } else {
+                dev
+            }
+        }
+        Err(e) => return Err(e),
+    };
+
+    let identifiers = identifiers_from_metadata(&mut device)?;
+    let key_description = key_desc_from_metadata(&mut device)?;
+    let key_description = match KeyDescription::from_system_key_desc(&key_description) {
+        Some(Ok(description)) => description,
+        _ => {
+            return Err(LibcryptErr::Other(format!(
+                "key description {} found on devnode {} is not a valid Stratis key description",
+                key_description,
+                physical_path.display()
+            )));
+        }
+    };
+    let clevis_info = clevis_info_from_metadata(&mut device)?;
+    let name = name_from_metadata(&mut device)?;
+
+    let activated_path = match unlock_method {
+        Some(UnlockMethod::Keyring) => {
+            activate(&mut device, Either::Left(&key_description), &name)?
+        }
+        Some(UnlockMethod::Clevis) => activate(&mut device, Either::Right(physical_path), &name)?,
+        None => [DEVICEMAPPER_PATH, &name].iter().collect(),
+    };
+
+    Ok(Some(CryptHandle {
+        device,
+        path: BlockDevPath::node_with_children(
+            activated_path,
+            vec![BlockDevPath::leaf(physical_path.to_owned())],
+        ),
+        identifiers,
+        encryption_info: EncryptionInfo {
+            key_description,
+            clevis_info,
+        },
+        name,
+    }))
+}
+
+/// Create a device handle and load the LUKS2 header into memory from
+/// a phyiscal path.
```

Comment:
typo, "phyisical"

Reply:
Okay I added the fix commit in this PR. I'm going to close and squash.